client/src/pages/private/PSITable.js

```diff
@@ -0,0 +1,228 @@
+    'Vancouver Island',
+    'Northern',
+    'None',
+  ]);
+  const { auth } = AuthContext.useAuth();
+  const roles = useMemo(() => auth.user?.roles || [], [auth.user]);
+
+  const history = useHistory();
+  const location = useLocation();
+
+  const handleRequestSort = (event, property) => {
+    const isAsc = orderBy === property && order === 'asc';
+    setOrder(isAsc ? 'desc' : 'asc');
+    setOrderBy(property);
+  };
+
+  const fetchPSIs = async () => {
+    setLoadingData(true);
+    const response = await fetch(`${API_URL}/api/v1/psi`, {
+      headers: { Authorization: `Bearer ${store.get('TOKEN')}` },
+      method: 'GET',
+    });
+    if (response.ok) {
+      const data = await response.json();
+      const rowsData = data.map((row) => {
+        return {
+          ...row,
+          id: row.id,
+          cohorts: row.cohorts === undefined ? 0 : row.cohorts.length,
+          available_seats: row.cohorts === undefined ? 0 : row.cohorts.length,
+        };
+      });
+      setFetchedRows(rowsData);
+      setRows(rowsData.filter((row) => healthAuthorities.includes(row.health_authority)));
+    } else {
+      setRows([]);
+      setFetchedRows([]);
+    }
+    setLoadingData(false);
+  };
+
+  const handlePSICreate = async (psi) => {
+    const response = await fetch(`${API_URL}/api/v1/psi`, {
+      method: 'POST',
+      headers: {
+        Authorization: `Bearer ${store.get('TOKEN')}`,
+        Accept: 'application/json',
+        'Content-type': 'application/json',
+      },
+      body: JSON.stringify(psi),
+    });
+
+    if (response.ok) {
+      setActiveModalForm(null);
+      fetchPSIs();
+    } else {
+      const error = await response.json();
+      if (error.code) {
+        // Closes the menu on duplicate submission
+        if (error.code === '23505') {
+          setActiveModalForm(null);
+        }
+        openToast({
+          status: ToastStatus.Error,
+          message: error.error || response.error || response.statusText || 'Server error',
+        });
+      }
+    }
+  };
+
+  useEffect(() => {
+    setHealthAuthorities(
+      roles.includes('superuser') || roles.includes('ministry_of_health')
+        ? Object.values(regionLabelsMap)
+        : roles.map((loc) => regionLabelsMap[loc]).filter(Boolean)
+    );
+  }, [roles]);
+
+  const sort = (array) => _orderBy(array, [orderBy, 'operatorName'], [order]);
+
+  useEffect(() => {
+    fetchPSIs();
+    // This fetch PSIs is a dependency of this function. This needs to be reworked, but it is outside of the scope of the ticket
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [history, location]);
+
+  const defaultOnClose = () => {
+    setActiveModalForm(null);
+  };
+  return (
+    <>
+      <Dialog
+        title={`Create New Institute`}
+        open={activeModalForm != null}
+        onClose={defaultOnClose}
+      >
+        {activeModalForm === 'new-psi' && (
+          <NewPSIForm
+            initialValues={{
```

Comment:
NIT, but it might be good to have the default values for this form in a constants file.